src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -415,6 +415,13 @@ public function getModule($name)
             $this->cache[$name]['disk'] = $disk;
         }
 
+        if ($this->moduleProvider->isModuleMainClassValid($name)) {
+            require_once $php_file_path;
+
+            $module = \PrestaShop\PrestaShop\Adapter\ServiceLocator::get($name);
+            $attributes['warning'] = $module->warning;
+        }
```

Comment:
With this piece of code, we lose all the performance gain from the cache because we always have to instanciate the modules (